File: plateforme/offer-pro-service/src/main/java/tn/esprit/Dtos/ApplyOnOfferDto.java
package tn.esprit.Dtos;

import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.Entitys.*;

import javax.persistence.Column;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ApplyOnOfferDto {
    long id;
    String username;
    StatusApply statutApply;
    LocalDateTime applyAt;
    /*String coverLetter;
    int salaryDesired;
    LocalDate startDate;*/

    RecruitmentOfferDto recruitmentOffer;
    List<Answer> answers = new ArrayList<Answer>();
}

File: webOfferPro/src/app/models/offer_pro_model/ApplyOnOfferDto.ts
import { AccountDto } from "../user_model/AcountDto";
import { Answer } from "./Answer";
import { RecruitmentOfferDto } from "./RecruitmentOfferDto";
import { StatusApply } from "./StatusApply";

export class ApplyOnOfferDto {
    id! : number ;
    username! : string ;
    statutApply! : StatusApply;
    applyAt! : Date;
    
    
    //coverLetter! : string ;
    //salaryDesired! : number ;
    //startDate! : Date;


    recruitmentOffer : RecruitmentOfferDto =  new RecruitmentOfferDto();
    answers : Answer[]  = new Array<Answer>();
    accountDto : AccountDto   = new   AccountDto ();
}   